main.cpp

```diff
@@ -6,21 +6,26 @@
 #include "Rectangle.hpp"
 #include "Square.hpp"
 #include "Circle.hpp"
+#include <math.h>
+#include <memory>
+#include <initializer_list>
 
 using namespace std;
 
-typedef vector<Shape*> Collection;
+using Collection = vector<std::shared_ptr<Shape>>;
 
-bool sortByArea(Shape* first, Shape* second)
+bool sortByArea(std::shared_ptr<Shape> first, std::shared_ptr<Shape> second)
 {
-    if(first == NULL || second == NULL)
-    {
-        return false;
-    }
-    return (first->getArea() < second->getArea());
+    [=](bool){
+        if(first == nullptr || second == nullptr)
+        {
+            return false;
+        }
+        return (first->getArea() < second->getArea());
+    };
 }
```

Comment:
to napewno nie kompiluje sie

Reply:
It's compiling -> tested, but i'm working on that to remove that function and replace it lambda expression.

Reply:
Function sortByArea replaced to:

    std::sort(shapes.begin(), shapes.end(),
        [&](auto& first, auto& second)->bool{
            if (first == nullptr || second == nullptr)
                return false;
            return first->getArea() < second-> getArea();
    });